lib/elastic_searcher.rb

```diff
@@ -13,24 +25,25 @@ def search
     )
     result.response # ES query is triggered here to allow fallback. avoids lazy loading done in the view
     [nil, result]
-  rescue Faraday::ConnectionFailed, Faraday::TimeoutError, Searchkick::Error, OpenSearch::Transport::Transport::Error => e
-    result = Rubygem.legacy_search(@query).page(@page)
-    [error_msg(e), result]
+  rescue StandardError => e
+    [error_msg(e), nil]
   end
 
   def api_search
     result = Rubygem.searchkick_search(body: search_definition(for_api: true).to_hash, page: @page, per_page: Kaminari.config.default_per_page,
 load: false)
     result.response["hits"]["hits"].pluck("_source")
-  rescue Faraday::ConnectionFailed, Faraday::TimeoutError, Searchkick::Error, OpenSearch::Transport::Transport::Error
-    Rubygem.legacy_search(@query).page(@page)
+  rescue Searchkick::InvalidQueryError
+    raise InvalidQueryError
```

Comment:
use `error_msg` so we can track errors

Reply:
We don't track errors for InvalidQuery, but I'll ensure we track error in API for the rest.